prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -610,9 +611,39 @@ ssize_t rxr_pkt_init_read_tagrtm(struct rxr_ep *ep,
  *     handle_sent() functions
  */
 
-/*
- *         rxr_pkt_handle_eager_rtm_sent() is empty and is defined in rxr_pkt_type_req.h
+/**
+ * @brief handler for an eager rtm packet has been sent
+ *
+ * @param[in]	ep		end point
+ * @param[in]	pkt_entry	eager rtm packet entry
  */
+void rxr_pkt_handle_eager_rtm_sent(struct rxr_ep *ep,
+				    struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_tx_entry *tx_entry;
+
+	/*
+	 * we cannot write completion here if:
+	 *
+	 *  1. zero copy send is being used. In this case, app's buffer
```

Comment:
This is definitely the wrong way to look at this.  We *can* write an early completion IF AND ONLY IF 1) we buffered the send, so that we can maintain the send completion buffer semantics AND the caller wants the provider to manage retries.  Otherwise, it would be erroneous to early return.

Reply:
Changed the comment and implementation to write completion only if two conditions are met.

Reply:
Removed this patch in new revision.